flake.nix

```diff
@@ -0,0 +1,184 @@
+# Copyright 2022 The cert-manager Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+{
+  description = ''
+    cert-manager/csi-lib
+
+    Go libary for building CSI drivers to deliver cert-manager signed
+    certificate key pairs to Kubernetes workloads.
+  '';
+
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
+
+  outputs = { self, nixpkgs, flake-utils }:
+    flake-utils.lib.eachDefaultSystem (system:
+      let
+        pkgs = import nixpkgs { inherit system; overlays = [ (import ./hack/nix/overlay-kind.nix) ]; };
+
+        # We only source go files to have better cache hits when actively
+        # working on non-go files.
+        src = pkgs.lib.sourceFilesBySuffices ./. [ ".go" "go.mod" "go.sum" ];
```

Comment:
What does this do? If we depend on say, a yaml file or some other testdata in our tests, does this not match them/what does it do? 👀

Reply:
This gets all the files under `./.` with the name prefix `.go`, `go.mod`, and `go.sum` (all go files).

That is correct- if a test depended on a non-file then we would need to use a different src strategy or not bother with the filtering at all if we didn't think it was worth it.